### PR TITLE
Add features for operationbuilder

### PR DESCRIPTION
**Pull request by cymqqqq**
new features for operation builder

### CODE REVIEW

**Comment by dr-orlovsky** [comment]
can you please rebase this on master removing all duplicated commits?

File: src/interface/iimpl.rs [hunk 1]
```diff
@@ -251,6 +251,33 @@ impl IfaceImpl {
...
+    }
+
+    pub fn extension_type(&self, name: &FieldName) -> Option<ExtensionType> {
+        self.extensions
+            .iter()
+            .find(|nt| &nt.name == name)
+            .map(|nt| nt.id)
+    }
+
+    pub fn valencies_type(&self, name: &FieldName) -> Option<ValencyType> {
```
Comment (dr-orlovsky) [inline src/interface/iimpl.rs:268]: This should be a singular: `valency_`

File: src/interface/iimpl.rs [hunk 2]
```diff
@@ -251,6 +251,33 @@ impl IfaceImpl {
...
+        self.valencies
+            .iter()
+            .find(|nt| &nt.name == name)
+            .map(|nt| nt.id)
+    }
+
+    pub fn valencies_name(&self, id: ValencyType) -> Option<&FieldName> {
```
Comment (dr-orlovsky) [inline src/interface/iimpl.rs:275]: Ibid

**Review by dr-orlovsky — CHANGES_REQUESTED** [review]

File: src/interface/iimpl.rs (hunk 1 above)
Comment (cymqqqq) [reply src/interface/iimpl.rs]: done

File: src/interface/iface.rs [hunk 3]
```diff
@@ -119,6 +119,14 @@ pub struct ValencyIface {
     pub multiple: bool,
 }
 
+impl ValencyIface {
+    pub fn any(req: Req) -> Self {
```
Comment (dr-orlovsky) [inline src/interface/iface.rs:123]: I am not sure I understand what is the purpose of this function. It also seems to be named incorrectly (it does something different to its name).

Comment (cymqqqq) [reply src/interface/iface.rs:123]: ok, maybe I misunderstand the purpose of this function, I will delete it.

**Review by dr-orlovsky — APPROVED** [review]

**Pull request merged**